app/models/user.rb

```diff
@@ -0,0 +1,36 @@
+require 'bcrypt'
+
+class User
+
+  include DataMapper::Resource
+
+  has n, :messages
+
+  property :id, Serial
+  property :email, String, unique: true, message: 'This email is already taken'
+  property :username, String
+  property :name, String
+  property :password_digest, Text
+
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_uniqueness_of :email
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.